api/specs.proto

```diff
@@ -364,3 +368,13 @@ message SecretSpec {
 	// Data is the secret payload - the maximum size is 500KB (that is, 500*1024 bytes)
 	bytes data = 2;
 }
+
+// ConfigSpec specifies user-provided configuration files.
+message ConfigSpec {
+	Annotations annotations = 1 [(gogoproto.nullable) = false];
+
+	// Data is the config payload - the maximum size is 500KB (that is, 500*1024 bytes)
```

Comment:
TBH I haven't put much thought into the payload of a config yet. So far I'm blindly copying secrets and I had been planning to come back to that. This is something we should figure out right now. Does it contain a single file? An array of files? A tarball? A choice between different options?

Any thoughts? cc @aluzzardi @dhiltgen